Create registration form

shhhQuiettt opened the pull request:
Add RegisterForm component
HTML CSS
Validation funtions in validation.js

shhhQuiettt committed:
Create registration form

Add RegisterForm component
HTML CSS
Validation funtions in validation.js

wojbog commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+            // console.log(user)
+
+            if  (err.length!==0) { //make invalid red etc.
+                console.log("dup")
```
niepotrzebny log

akruszewski replied:
Do repo wypychamy tylko produkcyjny kod. To nie wyglada na taki, do usuniecia.

wojbog reviewed (commented):
(jako niedoświadczony) wydaję mi się że jest ok

akruszewski commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+const RegisterForm = () => {
+
+    const [user, setUser] = useState({
+        "name": "Adam",
```
Wciecie jest off.

shhhQuiettt commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+                })
+            }else {
+
+            axios.post("https://localhost:3001/registration",
```
Trzeba zmienić na /newUser bo tak jest na serwerze, albo zmienić  na serwerze

akruszewski replied:
proponuje `createUser`, endpointy najlepiej jako nazwy akcji ktore wykonuja.

akruszewski commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+
+const RegisterForm = () => {
+
+    const [user, setUser] = useState({
```
State should be initialized empty.

akruszewski commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+        if(!passDiff &&passStr){
+            
+            let err = validate(user)
+            // console.log(user)
```
Nie wypychamy jako produkcyjnego kodu zakomentowanych lini.

akruszewski commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+                setInvalid(true)
+                err.forEach((id)=> {
+                    let invalid = document.getElementById(id)
+                    invalid.style.backgroundColor = "#ff9999"
```
Lepiej byloby to zrobic jako warunkowe wyswietlenie komponentu.

akruszewski commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+                })
+            }else {
+
+            axios.post("https://localhost:3001/registration",
```
Host aplikacji powinien byc zmienna, wtedy latwiej bedzie go zmienic.

akruszewski replied:
Komunikacje z api najlepiej zrobic w osobnym module serwisowym, np service.js, a pozniej, kiedy zrobi sie wiecej kodu podzielic je na osobne pliki, w zaleznosci od domeny, np. user, field, itp.

akruszewski commented on src/validation.js:
```diff
@@ -0,0 +1,60 @@
+
+ export function validate(usr) { //returns array of invalid keys
```
Najlepiej validatory umiescic w serwisie.

wojbog commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,108 @@
+                    event.target.value!==user.pass ? setPassDiff(true) : setPassDiff(false)}
+                 } required/>
+            
+            {passDiff && <p className="invalid">Hasła się nie zgadzają!</p>}
```
komunikat nie znika gdy poprawi się pole pass zamiast pass-confirmation, dodać sprawdzanie do pass

shhhQuiettt committed:
Change invalid fields handling and create service.js

Cleaned code
CreateUser and validation functions are in service.js now
Invalid fields are marked through conditional classes now
More comments added

shhhQuiettt committed:
Add Tests direcory

Add test for RegisterForm

akruszewski commented on src/service.js:
```diff
@@ -0,0 +1,71 @@
+            ).catch(error => {console.log("registration error: ", error)})
+}
+
+export function validate(usr) { //returns array of invalid keys
```
Nazwalbym to validateUser, zeby bylo wiadomo co to wlasciwie waliduje + jakas dokumentacja by sie przydala.

akruszewski commented on src/service.js:
```diff
@@ -0,0 +1,71 @@
+const HOST = "127.0.0.1"
+const PORT = "3001"
+
+export function createUser(user) {
```
Tutaj tez potrzebna jest dokumentacja funkcji.

akruszewski commented on src/components/RegisterForm.js:
```diff
@@ -0,0 +1,138 @@
+                 } data-testid="pass-confirmation" required/>
+            
+            {passDiff && <p className="invalid">Hasła się nie zgadzają!</p>}
+            {invalid && <p className="invalid">Jedno lub więcej pól jest nieprawidłowych!</p>}
```
To jest za mało jezeli chodzi o user expirience. Kazde pole powinno miec info walidacyjne, co konkretnie jest nie tak.

shhhQuiettt committed:
Create Input component, add tests and animations

Inputs and field-invalid-messages are now in Input component
RegisterForm test added
Animation for errors added
